homeassistant/components/qnap_qsw/binary_sensor.py

```diff
@@ -0,0 +1,106 @@
+"""Support for the QNAP QSW binary sensors."""
+from __future__ import annotations
+
+from dataclasses import dataclass
+from typing import Final
+
+from aioqsw.const import (
+    QSD_ANOMALY,
+    QSD_FIRMWARE_CONDITION,
+    QSD_MESSAGE,
+    QSD_PRODUCT,
+    QSD_SYSTEM_BOARD,
+)
+
+from homeassistant.components.binary_sensor import (
+    BinarySensorDeviceClass,
+    BinarySensorEntity,
+    BinarySensorEntityDescription,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.helpers.entity import EntityCategory
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from . import QswEntity
+from .const import ATTR_MESSAGE, DOMAIN
+from .coordinator import QswUpdateCoordinator
+
+
+@dataclass
+class QswBinarySensorEntityDescription(BinarySensorEntityDescription):
+    """A class that describes QNAP QSW binary sensor entities."""
+
+    attributes: dict[str, list[str]] | None = None
+    subkey: str = ""
+
+
+BINARY_SENSOR_TYPES: Final[tuple[QswBinarySensorEntityDescription, ...]] = (
+    QswBinarySensorEntityDescription(
+        attributes={
+            ATTR_MESSAGE: [QSD_FIRMWARE_CONDITION, QSD_MESSAGE],
+        },
+        device_class=BinarySensorDeviceClass.PROBLEM,
+        entity_category=EntityCategory.DIAGNOSTIC,
+        key=QSD_FIRMWARE_CONDITION,
+        name="Anomaly",
+        subkey=QSD_ANOMALY,
+    ),
+)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant, entry: ConfigEntry, async_add_entities: AddEntitiesCallback
+) -> None:
+    """Add QNAP QSW binary sensors from a config_entry."""
+    coordinator: QswUpdateCoordinator = hass.data[DOMAIN][entry.entry_id]
+    async_add_entities(
+        QswBinarySensor(coordinator, description, entry)
+        for description in BINARY_SENSOR_TYPES
+        if (
+            description.key in coordinator.data
+            and description.subkey in coordinator.data[description.key]
+        )
+    )
+
+
+class QswBinarySensor(QswEntity, BinarySensorEntity):
+    """Define a QNAP QSW binary sensor."""
+
+    entity_description: QswBinarySensorEntityDescription
+
+    def __init__(
+        self,
+        coordinator: QswUpdateCoordinator,
+        description: QswBinarySensorEntityDescription,
+        entry: ConfigEntry,
+    ) -> None:
+        """Initialize."""
+        super().__init__(coordinator, entry)
+        self._attr_name = (
+            f"{self.get_device_value(QSD_SYSTEM_BOARD, QSD_PRODUCT)} {description.name}"
```

Comment:
You could save a line if you made `self.get_device_value(QSD_SYSTEM_BOARD, QSD_PRODUCT)` a variable

Reply:
Done in 85c0cad84ab399aafadd0c1c1e2c9de6365e33af